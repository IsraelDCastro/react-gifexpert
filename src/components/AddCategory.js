import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (input) => {
    setInputValue(input.target.value)
  }
  const handleSubmit = (form) => {
    form.preventDefault();
    if ( inputValue.trim().length > 2) {
      setCategories( (categories) => [inputValue, ...categories] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}


