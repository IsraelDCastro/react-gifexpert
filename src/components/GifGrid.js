import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h4 className="animate__animated animate__fadeIn">{ category }</h4>
      { loading && <p className="animate__animated animate__flash">"Loading..."</p> }
      <div className="card-grid">
        { images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
