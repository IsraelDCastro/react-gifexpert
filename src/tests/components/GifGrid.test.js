import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests in <GifGrid />", () => {
  const category = "One piece";

  test("must be rendered correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("must show item when images loaded useFetchGifs", () => {
    const images = [{
      id: 'abc',
      url: "https://localhost:8080/something.jpeg",
      title: "Something else"
    }]
    useFetchGifs.mockReturnValue({
      data: images,
      loading: false
    })
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(images.length);
  });
});
