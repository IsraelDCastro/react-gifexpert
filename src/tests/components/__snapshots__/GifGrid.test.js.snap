// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> must be rendered correctly 1`] = `
<Fragment>
  <h4
    className="animate__animated animate__fadeIn"
  >
    One piece
  </h4>
  <p
    className="animate__animated animate__flash"
  >
    "Loading..."
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests in <GifGrid /> must show item when images loaded useFetchGifs 1`] = `
<Fragment>
  <h4
    className="animate__animated animate__fadeIn"
  >
    One piece
  </h4>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Something else"
      url="https://localhost:8080/something.jpeg"
    />
  </div>
</Fragment>
`;
